libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -690,6 +691,11 @@ bool AC_WPNav::advance_wp_target_along_track(float dt)
                 Vector3f dist_to_dest = (curr_pos - Vector3f(0,0,terr_offset)) - _destination;
                 if( dist_to_dest.length() <= _wp_radius_cm ) {
                     _flags.reached_destination = true;
+                    //hal.console->println("reached");
+                    if( _wp_complete_fn_t ){
+                        //hal.console->println("call complete fn");
+                        _wp_complete_fn_t(65535);
```

Comment:
I just think that I should use something different to tell that is a guided waypoint reach event instead of auto waypoint reach, so I use this magic number as a flag, I should have added comment, sorry